pkg/services/util/sign.go

```diff
@@ -195,31 +172,21 @@ func (s *SignService) HandleUnaryRequest(ctx context.Context, req any, handler U
 	}
 
 	if err != nil {
-		if !statusSupported {
-			return nil, err
-		}
-
 		resp = blankResp()
 
 		setStatusV2(resp, err)
 	}
 
 	// sign the response
-	if err = signResponse(s.key, resp, statusSupported); err != nil {
-		return nil, err
+	if err = signature.SignServiceMessage(s.key, resp); err != nil {
+		// We can't pass this error as status code since response will be unsigned.
+		// Isn't expected in practice, so panic is ok here.
+		panic(err)
```

Comment:
Brutal. We can log and close the connection.

Reply:
#2848